hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataSync.java

```diff
@@ -0,0 +1,529 @@
+      fixWriteStatusForExternalPaths(writeStat, commitTime);
+      tgtTableCommitMetadata.addWriteStat(partition, writeStat);
+    }
+
+    WriteOperationType operationType = sourceCommitMetadata.getOperationType();
+    tgtTableCommitMetadata.setOperationType(operationType);
+    tgtTableCommitMetadata.setCompacted(false);
+    if (!sourceCommitMetadata.getExtraMetadata().isEmpty()) {
+      sourceCommitMetadata.getExtraMetadata().forEach(tgtTableCommitMetadata::addMetadata);
+    }
+    return tgtTableCommitMetadata;
+  }
+
+  private HoodieReplaceCommitMetadata buildReplaceCommitMetadata(HoodieReplaceCommitMetadata srcReplaceMetadata, String commitTime) {
+    HoodieReplaceCommitMetadata tgtReplaceCommitMetadata = new HoodieReplaceCommitMetadata();
+
+    for (HoodieWriteStat writeStat : srcReplaceMetadata.getWriteStats()) {
+      String partition = writeStat.getPartitionPath();
+      fixWriteStatusForExternalPaths(writeStat, commitTime);
+      tgtReplaceCommitMetadata.addWriteStat(partition, writeStat);
+    }
+
+    WriteOperationType operationType = srcReplaceMetadata.getOperationType();
+    tgtReplaceCommitMetadata.setOperationType(operationType);
+    tgtReplaceCommitMetadata.setCompacted(false);
+    tgtReplaceCommitMetadata.setPartitionToReplaceFileIds(srcReplaceMetadata.getPartitionToReplaceFileIds());
+    if (!srcReplaceMetadata.getExtraMetadata().isEmpty()) {
+      srcReplaceMetadata.getExtraMetadata().forEach(tgtReplaceCommitMetadata::addMetadata);
+    }
+    return tgtReplaceCommitMetadata;
+  }
+
+  private void fixWriteStatusForExternalPaths(HoodieWriteStat writeStat, String commitTime) {
+    String originalPath = writeStat.getPath();
+    writeStat.setPath(ExternalFilePathUtil.appendCommitTimeAndExternalFileMarker(originalPath, commitTime));
+  }
+
+  private HoodieWriteConfig getWriteConfig(
+      Schema schema,
+      HoodieTableMetaClient metaClient,
+      String basePathOverride) {
+    return getWriteConfig(schema, Integer.parseInt(HoodieArchivalConfig.MAX_COMMITS_TO_KEEP.defaultValue()),
+        HoodieMetadataConfig.COMPACT_NUM_DELTA_COMMITS.defaultValue(), 120, metaClient,
+        basePathOverride);
+  }
+
+  private HoodieWriteConfig getWriteConfig(
+      Schema schema,
+      int numCommitsToKeep,
+      int maxNumDeltaCommitsBeforeCompaction,
+      int timelineRetentionInHours,
+      HoodieTableMetaClient metaClient,
+      String basePathOverride) {
+    Properties properties = new Properties();
+    properties.setProperty(HoodieMetadataConfig.AUTO_INITIALIZE.key(), "false");
+    return HoodieWriteConfig.newBuilder()
+        .withIndexConfig(HoodieIndexConfig.newBuilder().withIndexType(INMEMORY).build())
+        .withPath(metaClient.getBasePathV2().toString())
+        .withPopulateMetaFields(metaClient.getTableConfig().populateMetaFields())
+        .withEmbeddedTimelineServerEnabled(false)
+        .withSchema(schema == null ? "" : schema.toString())
+        .withArchivalConfig(
```

Comment:
oh for target table, cleaner is a no op right. 
So, only archival matters. 
lets ensure we take inputs from spark submit command use it. 
may be by default, having 120 commits in activetimeline is fine.